Source/WebKit/UIProcess/ios/ProcessStateMonitor.mm

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright (C) 2022 Apple Inc. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS''
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+ * THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS
+ * BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#import "config.h"
+#import "ProcessStateMonitor.h"
+
+#if PLATFORM(IOS_FAMILY)
+
+#import "Logging.h"
+#import "NetworkProcessMessages.h"
+#import "ProcessAssertion.h"
+#import "RunningBoardServicesSPI.h"
+
+namespace WebKit {
+
+static constexpr Seconds prepareForSuspensionDuration = 15_s;
+
+ProcessStateMonitor::ProcessStateMonitor(Function<void(bool)>&& becomeSuspendedHandler)
+    : m_becomeSuspendedHandler(WTFMove(becomeSuspendedHandler))
+    , m_suspendTimer(RunLoop::main(), this, &ProcessStateMonitor::suspendTimerFired)
+{
+    RELEASE_ASSERT(RunLoop::isMain());
+
```

Comment:
I think we want release logging whenever this object gets constructed / destructed. Given that we rely on visibility notifications to construct / destroy this object, I think it will help with debugging if we find issues in the wild.